### PR TITLE
Build bigger better

### PR DESCRIPTION
**Pull request by PederHundeboel**
buildtool now intercepts client packets and stops the from being sent to the server, aswell as ignores server position updates(this is only the case while the tool is building)

added a delay every 3rd buildstep, as to ensure the server doesn't discard packages.

added buildqueue for queueing actions that should be sent to server, whilst client input is intercepted.

### CODE REVIEW

File: src/main/java/net/fabricmc/unbreakable/BuildTool/BuildQueue.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    public enum ActionType {
+        PlaceBlock,
+        MovePosition,
+        None
+    }
+
+    private static ConcurrentLinkedQueue<BuildAction> _actionQueue = new ConcurrentLinkedQueue<>();
+
+    public BuildQueue(){
+
```
Comment (MichaelLykkeStrand) [inline src/main/java/net/fabricmc/unbreakable/BuildTool/BuildQueue.java:22]: Could use lombok :D @NoArgsConstructor

File: src/main/java/net/fabricmc/unbreakable/BuildTool/BuildQueue.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    public Packet GetNextPacket(){
+        return _actionQueue.poll().packet;
+    }
+
+    private static class BuildAction {
+        public Packet packet;
+        public ActionType type;
+        public BuildAction(Packet p, ActionType t){
+            packet = p;
```
Comment (MichaelLykkeStrand) [inline src/main/java/net/fabricmc/unbreakable/BuildTool/BuildQueue.java:43]: Min mor siger det ikke er matematik

File: src/main/java/net/fabricmc/unbreakable/BuildTool/BuilderTool.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+            isBuilding = true;
+            buildNext = false;
+            var con = client.getNetworkHandler();
+
+            ArrayList<Vec3d> offsets = schematic.GetOffsets();
+
+            BuilderTask builderTask = new BuilderTask(offsets, con, client);
+            Thread t = new Thread(builderTask);
+            t.start();
+
```
Comment (MichaelLykkeStrand) [inline src/main/java/net/fabricmc/unbreakable/BuildTool/BuilderTool.java:32]: Big gap in teeth

File: src/main/java/net/fabricmc/unbreakable/mixin/buildtool/MovementPacketC2SMixin.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+@Mixin(ClientPlayNetworkHandler.class)
+public abstract class MovementPacketC2SMixin {
+	@Shadow
+	private ClientConnection connection;
+	BuildQueue bq = new BuildQueue();
+	@Inject(method = "sendPacket(Lnet/minecraft/network/Packet;)V", at = @At("HEAD"), cancellable = true)
+	private void onSendPacket(Packet<?> packet, CallbackInfo ci) {
+
+		if (BuilderTool.isBuilding)
```
Comment (MichaelLykkeStrand) [inline src/main/java/net/fabricmc/unbreakable/mixin/buildtool/MovementPacketC2SMixin.java:24]: Kan være sin egen metode :D

**Review by MichaelLykkeStrand — APPROVED** [review]
Looks good

File: src/main/java/net/fabricmc/unbreakable/mixin/buildtool/MovementPacketC2SMixin.java (hunk 4 above)
Comment (PederHundeboel) [reply src/main/java/net/fabricmc/unbreakable/mixin/buildtool/MovementPacketC2SMixin.java]: yes it be strategy

**Review by MichaelLykkeStrand — APPROVED** [review]
Ok

**Pull request merged**